dspace-swordv2/src/main/java/org/dspace/sword2/CollectionListManagerDSpace.java

```diff
@@ -67,134 +77,145 @@ public Feed listCollectionContents(IRI colIRI, AuthCredentials authCredentials,
+                    .getProperty("swordv2-server", "title.field"));
+            title = title == null ? "Untitled" : title;
+            entry.setTitle(title);
+            entry.addLink(urlManager.getContentUrl(item).toString(),
+                    "edit-media");
+        }
+
+        return feed;
+    }
+
+    private List<Item> listItems(SwordContext sc, Collection collection,
+            SwordConfiguration swordConfig)
+            throws DSpaceSwordException
+    {
+        try
+        {
+            EPerson person = sc.getOnBehalfOf() != null ?
+                    sc.getOnBehalfOf() :
+                    sc.getAuthenticated();
+            List<Item> collectionItems = new ArrayList<Item>();
+
+            // first get the ones out of the archive
+            Iterator<Item> items = itemService
+                    .findBySubmitter(sc.getContext(), person);
+            while (items.hasNext())
+            {
+                Item item = items.next();
+                List<Collection> cols = item.getCollections();
+                for (Collection col : cols)
+                {
+                    if (col.getID().equals(collection.getID()))
+                    {
+                        collectionItems.add(item);
+                        break;
+                    }
+                }
+            }
+
+            // now get the ones out of the workspace
+            List<WorkspaceItem> wsis = workspaceItemService
+                    .findByEPerson(sc.getContext(), person);
+            for (WorkspaceItem wsi : wsis)
+            {
                 Item item = wsi.getItem();
 
                 // check for the wsi collection
                 Collection wsCol = wsi.getCollection();
-                if (wsCol.getID() == collection.getID())
+                if (wsCol.getID().equals(collection.getID()))
                 {
                     collectionItems.add(item);
                 }
 
                 // then see if there are any additional associated collections
-				Collection[] cols = item.getCollections();
-				for (Collection col : cols)
-				{
-					if (col.getID() == collection.getID())
-					{
-						collectionItems.add(item);
-						break;
-					}
-				}
-			}
-
-			// finally get the ones out of the workflow
-			BasicWorkflowItem[] wfis = BasicWorkflowItem.findByEPerson(sc.getContext(), person);
-			for (BasicWorkflowItem wfi : wfis)
-			{
-				Item item = wfi.getItem();
+                List<Collection> cols = item.getCollections();
+                for (Collection col : cols)
+                {
+                    if (col.getID().equals(collection.getID()))
+                    {
+                        collectionItems.add(item);
+                        break;
+                    }
+                }
+            }
+
+            // finally get the ones out of the workflow
+            List<BasicWorkflowItem> wfis = basicWorkflowItemService
+                    .findByOwner(sc.getContext(), person);
```

Comment:
This is a small mistake (which I forgot to document until now: https://wiki.duraspace.org/display/DSPACE/DSpace+Service+based+api%3A+API+Changelist#DSpaceServicebasedapi:APIChangelist-WorkflowItemrefactoredinto:Interface(oldusagemovedto:BasicWorkflowItem)). This should use the "WorkflowServiceFactory.getInstance().getWorkflowItemService().findBySubmitter() method. This will create an abstraction from the workflow & return WorkflowItems which also have a getCollection() & getItem() method.